dialer_test.go

```diff
@@ -126,9 +126,7 @@ func TestDialWithAdminAPIErrors(t *testing.T) {
 	stop := mock.StartServerProxy(t, inst)
 	defer func() {
 		stop()
-		if err := cleanup(); err != nil {
-			t.Fatalf("%v", err)
-		}
+		cleanup()
```

Comment:
nit: Would it be better to use `_ := cleanup()` to indicate we know if returns an error (we just explicitly don't care)